Dockerfile

```diff
@@ -57,6 +57,11 @@ ENV SHELL /bin/bash
 RUN yes |ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install)"
 RUN brew doctor || true
 
+# Manually install protobuf with --no-binary flag before installing protoc
+RUN brew install python
```

Comment:
That is happening anyway when we run `brew install --with-plugins grpc` - we just need to preempt it. I tried to not do this, but then `pip` is stuck at a very early version (which doesn't support the --no-binary arg) and `pip install --upgrade pip` fails (I am not really sure why). Running `pip --version` before and after the `brew install --with-plugins grpc` command shows that it is doing the work of upgrading pip, as part of installing `python`. We move that work to the `brew install python` command.

This could easily be me not understanding docker correctly, and there is something else I should be doing to upgrade pip.

Alternatively, we could try manually uninstalling and reinstalling the protobuf pip package after `brew install --with-plugins grpc`